src/include/execution_kernel.hpp

```diff
@@ -344,14 +360,43 @@ __global__ void executionKernel([[maybe_unused]] int rank /*for debug*/, T* inpu
 #else   // !defined(MSCCLPP_DEVICE_HIP)
   __syncthreads();
 #endif  // !defined(MSCCLPP_DEVICE_HIP)
+
   localPlan = (DeviceExecutionPlan*)sharedMem;
   int nOperations = localPlan->nOperations;
   Operation* operations = localPlan->operations;
   DeviceHandle<SmChannel>* smChannels = localPlan->channels.smChannels;
   DeviceHandle<SimpleProxyChannel>* proxyChannels = localPlan->channels.proxyChannels;
 
+#if defined(ENABLE_NPKIT)
```

Comment:
Can we put these into an inline function?

Reply:
Variable declarations are merged to the head of executor kernel.

For the following NpKit::CollectGpuEventShm calls, I feel a little bit awkward to wrap them into an inline function because they belong to different purposes are enabled by different macro definitions.